Added providers for ProfileManager & GroupsManager

alankley opened the pull request:
Updated all screens using them.
Deprecated TriggersHelper
Should UsersManager be deprecated? getThese() copied to profileManager

Commit:
Added providers for ProfileManager & GroupsManager
Updated all screens using them.
Deprecated TriggersHelper
Can we delete UsersManager? getThese() copied to profileManager

Samaritan1011001 commented on lib/screens/HomeScreen.dart:
```diff
@@ -28,8 +30,10 @@ class HomeScreenState extends State<HomeScreen> {
+
     return StreamBuilder<QuerySnapshot>(
-          stream: CloudFirebaseHelper.getLocationsStream(ProfileManager().profile.friends),
+          stream: CloudFirebaseHelper.getLocationsStream(profile.friends),
```
Instead of this call to `getLocationsStream` should be replaced with a call to stream in the profile manager and that should internally use `getLocationsStream` and stream data back here. 

Idea is to have the UI separate from any of the backend calls and so `CloudFirebaseHelper` should ideally never be imported in a UI file like `HomeScreen`. What are your thoughts on this? Let me know.

alankley replied:
I like you're thinking.  I did a search on CloudFirebaseHelper, and there is also a direct call in GroupsScreen that could be refactored to move to GroupsManager.  
I will make both of these changes.

Only thing I would suggest, perhaps adding a LocationsManager to manage the cloud calls to the Locations Table, and not mix this with Profile.  Imagine future features like notifications when friends reach a destination, or notification when a friend is within a certain distance/time away.  This logic wouldn't belong in ProfileManager.  What do you think?

Are you ok using Manager to describe these classes.  I was thinking Provider might be a good name as well.

Should I go ahead and convert ThemeManager to a provider?

Samaritan1011001 reviewed (changes requested):
Everything else seems good to me 👍

Commit:
Created LocationsManager to expose getLocationsStream
Removed usage of CloudFirebaseHelper from GroupsScreen in favor of GroupsManager

Commit:
Add bigger margin to map

Samaritan1011001 commented on lib/screens/HomeScreen.dart:
```diff
@@ -28,8 +32,10 @@ class HomeScreenState extends State<HomeScreen> {
+
     return StreamBuilder<QuerySnapshot>(
-          stream: CloudFirebaseHelper.getLocationsStream(ProfileManager().profile.friends),
+          stream: LocationsManager.getLocationsStream(profile.friends),
```
We need to access the `LocationsManager` the same way as `ProfileManager` above. That is, as a provider. This can be added in a different PR but wanted to note it.

alankley replied:
Done

Samaritan1011001 commented on lib/screens/AddGroupScreen.dart:
```diff
@@ -156,45 +155,49 @@ import 'SignInScreen.dart';
+class FriendRow extends StatelessWidget {
+  final ProfileManager profileMgr;
+  final friendIdx;
+  final AddGroupScreenState state;
```
I think we shouldn't pass state like this, instead pass the data we need. If we want to use the parent state for some reason, I think using a callback is better. 

This SO answer is clear: https://stackoverflow.com/a/50085960

alankley replied:
Done

Samaritan1011001 reviewed (commented):
Just two comments. It's okay if you want it to add those in a different PR, wanted to make a note.

Commit:
Replace passing state to FriendRow with passing callbackFunction and selected value

Commit:
Use LocationManager as Provider
